android-sdk/src/main/java/com/checkout/android_sdk/View/CardDetailsView.java

```diff
@@ -404,8 +404,10 @@ private void checkFullDate() {
                 mDataStore.getCardYear() != null &&
                 !CardUtils.isValidDate(mDataStore.getCardMonth(), mDataStore.getCardYear())) {
             mDataStore.setValidCardMonth(false);
-            ((TextView) mMonthInput.getSelectedView()).setError(getResources()
-                    .getString(R.string.error_expiration_date));
+           if(mMonthInput.getSelectedView()!=null){
```

Comment:
Formatting looks a bit odd, please check spaces.

Reply:
In addition to the suggestion from @deepesh-vasthimal-cko:

logic in this function is a bit messy, can we clean it up a bit by inverting the if statements and doing some early returns?

Reply:
optimise if conditions and clean the blank spaces.